engine/src/main/java/io/zeebe/engine/nwe/container/ProcessProcessor.java

```diff
@@ -143,16 +143,7 @@ public void onTerminated(
 
     if (parentWorkflowInstanceKey > 0) {
       // workflow instance is created by a call activity
-
-      // TODO (saig0): move responsibility to call activity (#4473)
-      final var parentElementInstance = stateBehavior.getParentElementInstanceContext(context);
-
-      if (parentElementInstance != null
-          && parentElementInstance.getIntent() == WorkflowInstanceIntent.ELEMENT_TERMINATING) {
-        // terminate the corresponding call activity
-
-        stateTransitionBehavior.transitionToTerminated(parentElementInstance);
-      }
+      stateTransitionBehavior.onElementTerminated(element, context);
```

Comment:
:+1: 

We should also call `onElementCompleted()` in `onCompleted()`.

Reply:
We shouldn't directly, it already calls `takeOutgoingSequenceFlows()` which in turn also calls `onElementCompleted()` if there are no outgoing sequence flows. So indirectly we already do

Reply:
I'm sorry for the confusion :sweat_smile:  

I mean that there is a remaining `TODO` in `ProcessProcessor#onComplete()`. I think that `CallActivityProcessor#onChildCompleted()` is currently no called.